### PR TITLE
doc: Refactor coverage documentation

### PR DESCRIPTION
**Pull request by prcr**
The [main page with instructions for setting up coverage](https://docs.codacy.com/coverage-reporter/) is getting too big, making it hard to follow through the complete sequence of instructions.

This pull request:

- Moves the instructions for more advanced scenarios to a dedicated page
- Tweaks a few references and admonitions so that they appear in the sequence that makes the most sense while users are configuring their setup

These improvements will give way to more detailed troubleshooting instructions for the most common coverage setup failure scenarios (see [CY-6263](https://codacy.atlassian.net/browse/CY-6263)).

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -140,7 +140,14 @@ As a last resort, you can also send the coverage data directly by calling one of
 
 ## 2. Uploading coverage data to Codacy {: id="uploading-coverage"}
 
-After having coverage reports set up for your repository, you must use Codacy Coverage Reporter to convert the reports to smaller JSON files and upload these files to Codacy. The recommended way to do this is using a CI/CD platform that automatically runs tests, generates coverage, and uses Codacy Coverage Reporter to upload the coverage report information for every push to your repository.
+After having coverage reports set up for your repository, you must use the Codacy Coverage Reporter to convert the reports to smaller JSON files and upload these files to Codacy.
```
Comment (mrfyda) [inline docs/index.md:143]:
```suggestion
After having coverage reports set up for your repository, you must use the Codacy Coverage Reporter to upload them to Codacy.
```

Comment (prcr) [reply docs/index.md:143]: Fair enough, that's a nice simplification! :ok_hand: :heart:

File: docs/index.md [hunk 2]
```diff
@@ -140,7 +140,14 @@ As a last resort, you can also send the coverage data directly by calling one of
...
+
+The recommended way to do this is by using a CI/CD platform that automatically runs tests, generates coverage, and uses the Codacy Coverage Reporter to upload the coverage report information for every push to your repository.
+
+!!! note "Alternative ways of running the Codacy Coverage Reporter"
+    The instructions on this page assume that you'll run a self-contained bash script `get.sh` to automatically download and run the most recent version of the Codacy Coverage Reporter.
```
Comment (mrfyda) [inline docs/index.md:148]: for those with security concerns, would it be worth mentioning the checksum mechanism to guarantee the integrity of the reporter? linking to the source of the get.sh file?

File: docs/index.md [hunk 3]
```diff
@@ -189,93 +196,15 @@ After having coverage reports set up for your repository, you must use Codacy Co
 
     Check the console output to validate that the Codacy Coverage Reporter **detected the correct commit UUID** and **successfully uploaded** the coverage data to Codacy. If you need help, [check the troubleshooting page](troubleshooting-common-issues.md) for solutions to the most common setup issues.
 
-    !!! tip
-        The self-contained script `get.sh` automatically downloads and runs the most recent version of Codacy Coverage Reporter. See [alternative ways of running Codacy Coverage Reporter](alternative-ways-of-running-coverage-reporter.md) for other ways of running Codacy Coverage Reporter, such as by installing the binary manually or using a GitHub Action or CircleCI Orb.
-
-        Be sure to also check the sections below for more advanced functionality while uploading the coverage data to Codacy.
+    !!! note
+        Be sure to also check the [instructions for more advanced scenarios](uploading-coverage-in-advanced-scenarios.md) while uploading the coverage data to Codacy, such as when you need to upload multiple coverage reports.
```
Comment (mrfyda) [inline docs/index.md:200]: would it be more giving examples of what these advanced use cases are? parallel tests, multiple languages, monorepos, ...

File: docs/index.md [hunk 4]
```diff
@@ -116,7 +116,7 @@ The following table contains example coverage tools that generate reports in for
 </tbody>
 </table>
 
-### Submitting coverage from unsupported report formats
+### Handling unsupported report formats
```
Comment (mrfyda) [inline docs/index.md:119]:
```suggestion
### Handling unsupported languages
```

Comment (mrfyda) [reply docs/index.md:119]: the table above appears to be more centered around languages

Comment (mrfyda) [reply docs/index.md:119]: and consider linking to the "unsupported languages" too

File: docs/index.md (hunk 2 above)
Comment (prcr) [reply docs/index.md:148]: That's a good idea! The link to the [source](https://github.com/codacy/codacy-coverage-reporter/blob/master/get.sh) is actually also missing here:

https://docs.codacy.com/coverage-reporter/alternative-ways-of-running-coverage-reporter/

File: docs/index.md (hunk 3 above)
Comment (prcr) [reply docs/index.md:200]: Good suggestion, this will help the users decide if they need to check out the advanced instructions or not.

File: docs/index.md (hunk 2 above)
Comment (prcr) [reply docs/index.md]: I reworked this a bit, I think it's quite better now. :nerd_face:

File: docs/index.md (hunk 4 above)
Comment (prcr) [reply docs/index.md:119]: To my eyes, the limiting factor here seems to be the coverage report formats that the Codacy Coverage Reporter is able to parse (the table above gives examples of the common report formats generated for each language).

But this may just be a semantic detail, and in the end, the coverage reporter converters that we're suggesting will most likely be needed when the developers are working on a language that isn't included in the table. :+1:
